src/Engine/Manager/RenderManager.cpp

```diff
@@ -203,82 +230,124 @@ void RenderManager::UpdateBufferSize() {
 void RenderManager::RenderWorldEntities(World& world, const glm::mat4& viewMatrix, const glm::mat4& projectionMatrix, Video::RenderSurface* renderSurface) {
     // Render from camera.
     renderer->StartRendering(renderSurface);
+    glm::mat4 lightViewMatrix;
+    glm::mat4 lightProjection;
+    float aspectRatio = static_cast<float>(shadowPass->GetShadowWidth()) / shadowPass->GetShadowHeight();
+
+    for (Component::SpotLight* spotLight : spotLights.GetAll()) {
+        if (spotLight->IsKilled() || !spotLight->entity->enabled)
+            continue;
+
+        if (spotLight->shadow) {
+            Entity* lightEntity = spotLight->entity;
+            lightViewMatrix = glm::inverse(lightEntity->GetModelMatrix());
+            lightProjection = glm::perspective(glm::radians(2.f * spotLight->coneAngle), aspectRatio, 0.5f, 50.0f);
```

Comment:
If the spotlight has a range or extent we may want to use that as the far plane instead of some arbitrary value.

Reply:
Ignored, can only use arbitary value for now